tests/test_gamma.py

```diff
@@ -363,3 +363,54 @@ def test_meta_data_exists(series_ifgs, parallel_ifgs):
         assert (s.meta_data[ifc.DATA_TYPE] == ifc.MULTILOOKED) or \
                (s.meta_data[ifc.DATA_TYPE] == ifc.MULTILOOKED_COH)
     assert i + 1 == 34
+
+
+class TestGammaBaselineRead:
+    """Tests the reading of initial and precise baselines"""
+
+    def setup_method(self):
+        init_path = join(GAMMA_TEST_DIR, '20160114-20160126_base_init.par')
+        self.init = gamma.parse_baseline_header(init_path)
+        prec_path = join(GAMMA_TEST_DIR, '20160114-20160126_base.par')
+        self.prec = gamma.parse_baseline_header(prec_path)
+
+    def test_prec_baseline_read(self):
+        """Test that the Precise baseline values are being read"""
+        exp_i = {'BASELINE_T': -0.0000026, 'BASELINE_C': -103.7427072,
+                 'BASELINE_N': 2.8130731, 'BASELINE_RATE_T': 0.0,
+                 'BASELINE_RATE_C': -0.0173538,
+                 'BASELINE_RATE_N': -0.0055098}
+
+        exp_p = {'BASELINE_T': 0.0, 'BASELINE_C': -103.8364725,
+                 'BASELINE_N': 2.8055662, 'BASELINE_RATE_T': 0.0,
+                 'BASELINE_RATE_C': -0.0182215,
+                 'BASELINE_RATE_N': -0.0065402}
+
+        # Precise values are read
+        assert self.prec != exp_i
+        assert self.prec == exp_p
+
+        # Initial values are ignored
+        assert self.init != exp_i
+        assert self.init != exp_p
+
+
+    def test_init_baseline_read(self):
+        """Test that the Initial baseline values are being read"""
+        exp_i = {'BASELINE_T': 0.6529765, 'BASELINE_C': -103.9065694,
+                 'BASELINE_N': 2.9253896, 'BASELINE_RATE_T': 0.0,
+                 'BASELINE_RATE_C': -0.0231786,
+                 'BASELINE_RATE_N': -0.0038703}
+
+        exp_p = {'BASELINE_T': 0.0, 'BASELINE_C': 0.0,
+                 'BASELINE_N': 0.0, 'BASELINE_RATE_T': 0.0,
+                 'BASELINE_RATE_C': 0.0, 'BASELINE_RATE_N': 0.0}
+
+        # Precise values are ignored
+        assert self.prec != exp_i
+        assert self.prec != exp_p
+
+        # Initial values are read
+        assert self.init == exp_i
+        assert self.init != exp_p
```

Comment:
for an equality check in a dict, only one element mismatch will result true. Do you need all  elements to be different? Or  is just one unequal element in the dict is enough here?

Reply:
Thanks for questioning @basaks. Because we are testing against hard-coded truths here (`exp_i` and `exp_p` values are derived from the files that are being read in the tests: `20160114-20160126_base_init.par` and `20160114-20160126_base.par`), it's either all values (i.e. 6) equal or no values equal, nothing in between. So I think it should be ok?